module_text_llm/module_text_llm/helpers/models/openai.py

```diff
@@ -0,0 +1,312 @@
+# Monkey patching end                                                   #
+#########################################################################
+
+
+def _use_azure_credentials():
+    openai.api_type = "azure"
+    openai.api_key = os.environ.get("LLM_AZURE_OPENAI_API_KEY")
+    openai.api_base = os.environ.get("LLM_AZURE_OPENAI_API_BASE")
+    # os.environ.get("LLM_AZURE_OPENAI_API_VERSION")
+    openai.api_version = "2022-06-01-preview"
+
+
+def _use_openai_credentials():
+    openai.api_type = "open_ai"
+    openai.api_key = os.environ.get("LLM_OPENAI_API_KEY")
+    openai.api_base = "https://api.openai.com/v1"
+    openai.api_version = None
+
+
+openai_available = os.environ.get("LLM_OPENAI_API_KEY") is not None
+azure_openai_available = os.environ.get("LLM_AZURE_OPENAI_API_KEY") is not None
+
+
+# This is a hack to make sure that the openai api is set correctly
+# Right now it is overkill, but it will be useful when the api gets fixed and we no longer
+# hardcode the model names (i.e. OpenAI fixes their api)
+@contextmanager
+def _openai_client(use_azure_api: bool, is_preference: bool):
+    """Set the openai client to use the correct api type, if available
+
+    Args:
+        use_azure_api (bool): If true, use the azure api, else use the openai api
+        is_preference (bool): If true, it can fall back to the other api if the preferred one is not available
+    """
+    if use_azure_api:
+        if azure_openai_available:
+            _use_azure_credentials()
+        elif is_preference and openai_available:
+            _use_openai_credentials()
+        elif is_preference:
+            raise EnvironmentError(
+                "No OpenAI api available, please set LLM_AZURE_OPENAI_API_KEY, LLM_AZURE_OPENAI_API_BASE and "
+                "LLM_AZURE_OPENAI_API_VERSION environment variables or LLM_OPENAI_API_KEY environment variable"
+            )
+        else:
+            raise EnvironmentError(
+                "Azure OpenAI api not available, please set LLM_AZURE_OPENAI_API_KEY, LLM_AZURE_OPENAI_API_BASE and "
+                "LLM_AZURE_OPENAI_API_VERSION environment variables"
+            )
+    else:
+        if openai_available:
+            _use_openai_credentials()
+        elif is_preference and azure_openai_available:
+            _use_azure_credentials()
+        elif is_preference:
+            raise EnvironmentError(
+                "No OpenAI api available, please set LLM_OPENAI_API_KEY environment variable or LLM_AZURE_OPENAI_API_KEY, "
+                "LLM_AZURE_OPENAI_API_BASE and LLM_AZURE_OPENAI_API_VERSION environment variables"
+            )
+        else:
+            raise EnvironmentError(
+                "OpenAI api not available, please set LLM_OPENAI_API_KEY environment variable"
+            )
+
+    # API client is setup correctly
+    yield
+
+
+def _get_available_deployments(openai_models: dict[str, List[str]], model_aliases: dict[str, str]):
```

Comment:
```suggestion
def _get_available_deployments(openai_models: Dict[str, List[str]], model_aliases: Dict[str, str]):
```
I usually prefer `dict`, but then you don't support Python 3.8. I'm not sure if we even want to, but only uppercasing all built-in types seems like a low price to pay for that, so I think we should keep them capitalized.

Reply:
Okay